contracts/manager/CrunaManager.sol

```diff
@@ -22,86 +22,129 @@ import {Deployed} from "../utils/Deployed.sol";
  * It is the only contract that can manage the NFT. It sets protectors and safe recipients,
  * plugs and manages services, and has the ability to transfer the NFT if there are protectors.
  */
-contract CrunaManager is Actor, CrunaManagerBase, Deployed {
+contract CrunaManager is GuardianInstance, Actor, CrunaManagerBase, Deployer {
   using ECDSA for bytes32;
   using Strings for uint256;
   using ExcessivelySafeCall for address;
 
   /**
    * @notice The array of all plugged services. The max length is set to 16 to avoid gas issues.
    */
-  PluginElement[] private _allPlugins;
+  bytes32[] private _allPlugins;
 
   /**
    * @notice The mapping of all services by key. A key is the combination of the nameId and the salt.
    */
-  mapping(bytes8 pluginKey => PluginConfig pluginDetails) private _pluginByKey;
+  mapping(bytes32 _pluginKey => PluginConfig pluginDetails) private _pluginByKey;
 
-  /// @dev see {IVersioned-version}
+  /**
+   * @notice Returns the version of the contract.
+   * The format is similar to semver, where any element takes 3 digits.
+   * For example, version 1.2.14 is 1_002_014.
+   */
   function version() external pure virtual override returns (uint256) {
     return 1_001_000;
   }
 
-  /// @dev see {ICrunaManager-pluginByKey}
-  function pluginByKey(bytes8 key) external view returns (PluginConfig memory) {
+  /**
+   * @dev It returns the configuration of a plugin by key
+   * @param key The key of the plugin
+   */
+  function pluginByKey(bytes32 key) external view returns (PluginConfig memory) {
     return _pluginByKey[key];
   }
 
-  /// @dev see {ICrunaManager-allPlugins}
-  function allPlugins() external view returns (PluginElement[] memory) {
+  /**
+   * @dev It returns the configuration of all currently plugged services
+   */
+  function allPlugins() external view returns (bytes32[] memory) {
     return _allPlugins;
   }
 
-  /// @dev see {ICrunaManager-pluginByIndex}
-  function pluginByIndex(uint256 index) external view returns (PluginElement memory) {
+  /**
+   * @dev It returns an element of the array of all plugged services
+   * @param index The index of the plugin in the array
+   */
+  function pluginByIndex(uint256 index) external view returns (bytes32) {
     if (index >= _allPlugins.length) revert IndexOutOfBounds();
     return _allPlugins[index];
   }
 
-  /// @dev see {ICrunaManager-migrate}
+  /**
+   * @dev During an upgrade allows the manager to perform adjustments if necessary.
+   * The parameter is the version of the manager being replaced. This will allow the
+   * new manager to know what to do to adjust the state of the new manager.
+   */
   function migrate(uint256 /* version */) external virtual override {
```

Comment:
migrate is a fairly generic word, maybe pluginMigrate

Reply:
Yes, but that is in the manager, so we do not risk any conflicts. The function in ICrunaService and ICrunaManagedService are the ones that must avoid conflicts.